lib/src/main/java/com/scalar/admin/kubernetes/Pauser.java

```diff
@@ -77,67 +77,163 @@ public PausedDuration pause(int pauseDuration, @Nullable Long maxPauseWaitTime)
           "pauseDuration is required to be greater than 0 millisecond.");
     }
 
-    TargetSnapshot target;
+    // Get pods and deployment information before pause.
+    TargetSnapshot targetBeforePause;
     try {
-      target = targetSelector.select();
+      targetBeforePause = getTarget();
     } catch (Exception e) {
       throw new PauserException("Failed to find the target pods to pause.", e);
     }
 
-    RequestCoordinator coordinator = getRequestCoordinator(target);
-
-    coordinator.pause(true, maxPauseWaitTime);
-
-    Instant startTime = Instant.now();
-
-    Uninterruptibles.sleepUninterruptibly(pauseDuration, TimeUnit.MILLISECONDS);
+    // Get RequestCoordinator of Scalar Admin to pause.
+    RequestCoordinator requestCoordinator;
+    try {
+      requestCoordinator = getRequestCoordinator(targetBeforePause);
+    } catch (Exception e) {
+      throw new PauserException("Failed to initialize the request coordinator.", e);
+    }
 
-    Instant endTime = Instant.now();
+    // Run pause operation.
+    PauseFailedException pauseFailedException = null;
+    try {
+      pauseInternal(requestCoordinator, pauseDuration, maxPauseWaitTime);
+    } catch (Exception e) {
+      pauseFailedException = new PauseFailedException("Pause operation failed.", e);
+    }
 
-    unpauseWithRetry(coordinator, MAX_UNPAUSE_RETRY_COUNT, target);
+    // Run unpause operation.
+    UnpauseFailedException unpauseFailedException = null;
+    try {
+      unpauseWithRetry(requestCoordinator, MAX_UNPAUSE_RETRY_COUNT);
+    } catch (Exception e) {
+      unpauseFailedException = new UnpauseFailedException("Unpause operation failed.", e);
+    }
 
+    // Get pods and deployment information after pause.
     TargetSnapshot targetAfterPause;
     try {
-      targetAfterPause = targetSelector.select();
+      targetAfterPause = getTarget();
     } catch (Exception e) {
       throw new PauserException(
           "Failed to find the target pods to examine if the targets pods were updated during"
               + " paused.",
           e);
```

Comment:
If pausing or unpausing fails, shouldn't we display a message about it here?

Reply:
Ah, I see. You're right. We need to return the exception with messages here if the pause or unpause fails (especially, the unpause failure is critical).

I will fix it. Thank you for pointing it out!

Reply:
Fixed in d3b54c092b47410dfc9383424f9b7b608df1df7e.